### PR TITLE
Article category selection

### PR DESCRIPTION
**Pull request by abdullahcalisir12**
Article category selection page created accoring to issue #182 and this PR is alson related to #242 

New Article buttons direction changed to route named ArticleCategorySelection.

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -84,6 +85,14 @@ const routes = [
       requiresAuth: true
     }
   },
+  {
+    path: '/articles/category_selection',
```
Comment (mort3za) [inline src/router/index.js:89]: Word separator should be dash `-`

**Comment by mort3za** [comment]
I get error 400 without showing it to user: `The email of 5/28/2020 had already sent`

**Comment by mort3za** [comment]
Do you use these lines? If step concept changed, names should be updated and remove any step in component names and all unused components should be removed too.

```
<ArticleAddStep2 :error="validationError" />
<ArticleAddStep3 :error="validationError" />
```

**Comment by mort3za** [comment]
Conceptually select-category should be a part of article-add, but seems you directly link to select-category view, then you lead user to article-add

**Comment by abdullahcalisir12** [comment]
> Do you use these lines? If step concept changed, names should be updated and remove any step in component names and all unused components should be removed too.
> 
> ```
> <ArticleAddStep2 :error="validationError" />
> <ArticleAddStep3 :error="validationError" />
> ```

yes I use, yes updating names makes sense

**Comment by mort3za** [comment]
> yep I did so, will it be a problem?

Not for now, after few months and developing more yes.  
If that don't take time more than 0.5h worth to change it.

**Comment by abdullahcalisir12** [comment]
> > yep I did so, will it be a problem?
> 
> Not for now, after few months and developing more yes.
> If that don't take time more than 0.5h worth to change it.

it will take more than 0.5h I think.

**Comment by mort3za** [comment]
ok ignore it

**Pull request merged**